spi/common/verifiable-credentials-spi/src/main/java/org/eclipse/edc/iam/verifiablecredentials/spi/model/VerifiableCredential.java

```diff
@@ -119,6 +119,10 @@ public DataModelVersion getDataModelVersion() {
         return dataModelVersion;
     }
 
+    public <T extends VerifiableCredential, B extends Builder<T, B>> Builder<T, B> toBuilder() {
+        return new Builder(this);
```

Comment:
we must be aware with this approach (that differs from the current `toBuilder` implementations we have e.g. in `Policy` or `Asset` instances, that the changes applied on the builder will affect the original object instance.
Maybe we should decide on a standard approach about builders, because the different approaches used currently could cause confusion.

Reply:
excellent point.
as far as i can see, both `Asset` and `Policy` _affect_ the original instance (i.e. modify some related objects), whereas the `toBuilders` implemented here directly modify the original instance. subtle but an important difference.

to me, the implementation here is more correct, because it directly operates on the original instance. 
as an addition, we could introduce a `copy()` method that creates a carbon-copy. WDYT?

Reply:
maybe we could state a clear decision and change all the builders appropriately and to document it, I'm not sure if there are other builders that behave as `Asset` and `Policy`, but I guess so

Reply:
i'll open an issue about this